announcement of the deprecation of put for bankfeed bank accounts (#529)
codatio/codat-docs: blog/231115-put-bank-feed-bank-accounts.md

---

> title: "Upcoming 2023-11-15: Deprecation of PUT bankFeedAccounts for bank feeds account creation"

Reviewer: ```suggestion
title: "Upcoming 2023-11-15: Deprecation of the _Bank feeds account creation_ endpoint"
```

date: "2023-06-30"
tags: ["Deprecation"]
draft: false
authors: jyorston
---

> On November 15, 2023, Codat will deprecate the `PUT bankFeedAccounts` endpoint and introduce a new `POST bankFeedAccounts` endpoint. Using the same schema, the new endpoint will allow the creation of a single new bank feed account only. It will return an error if the account ID already exists for the data connection.

Reviewer: ```suggestion
On November 15, 2023, Codat will deprecate the `PUT bankFeedAccounts` endpoint and introduce a new `POST bankFeedAccounts` endpoint instead. The new endpoint will only allow the creation of a single new bank feed account. 
```

> Effective from November 15, 2023, Codat will be deprecating the `PUT bankFeedAccounts` endpoint and introducing a new `POST bankFeedAccounts` endpoint. Whilst the schema will be the same, the new endpoint will only allow the creation of a single new bank feed account and will return an error if the ID already exists for the data connection.

Reviewer: ```suggestion
On November 15, 2023, Codat will deprecate the `PUT bankFeedAccounts` endpoint and introduce a new `POST bankFeedAccounts` endpoint. While the schema will be the same, the new endpoint will only allow the creation of a single new bank feed account. It will return an error if the ID already exists for the data connection.
```

<!--truncate-->

> Currently, the `PUT bankFeedAccounts` endpoint handles creation and updating of bank feed accounts. This mixed functionality has caused some complexity and occasional misuse, such as unintended updates to bank account numbers.

Reviewer: ```suggestion
Currently, the `PUT bankFeedAccounts` endpoint handles both the creation and updating of bank feed accounts. After November 15, 2023, it will only manage the creation of new bank feed accounts, providing a clearer and more functional API.
```

> After November 15, 2023, the POST bankFeedAccounts endpoint will only manage the creation of new bank feed accounts, providing clearer and more functional API.

Reviewer: ```suggestion
```

> After November 15, 2023, the POST bankFeedAccounts endpoint will only manage the creation of new bank feed accounts, providing clearer and more functional API.

Reviewer: ```suggestion
```

## Action required​

> This is a modification aimed at improving the clarity and reliability of our API. If you are utilizing the `PUT bankFeedAccounts` endpoint, you will need to update your workflows to use the new `POST bankFeedAccounts` endpoint for creating new bank feed accounts.

Reviewer: ```suggestion
This change improves the clarity and reliability of our API. If you use the `PUT bankFeedAccounts` endpoint, you will need to update your workflows to call the new `POST bankFeedAccounts` endpoint for creating new bank feed accounts instead.
```

> The `PUT bankFeedAccounts` endpoint will be flagged as deprecated on the the public OAS documentation and will be removed on November 15th 2023. We recommend planning the transition to the new endpoint as early as possible.

Reviewer: ```suggestion
```

## Expected impact if no action is taken​

> The `PUT bankFeedAccounts` endpoint will no longer be available after the deprecation period. If you continue to use this endpoint without transitioning to the new `POST bankFeedAccounts` endpoint, your requests for creating new bank feed accounts will fail.

Reviewer: ```suggestion
The `PUT bankFeedAccounts` endpoint will no longer be available after November 15, 2023. If you continue to use this endpoint, your requests for creating new bank feed accounts will fail.
```
